src/logic/collectibles/store/actions/fetchCollectibles.ts

```diff
@@ -5,16 +5,26 @@ import { getConfiguredSource } from 'src/logic/collectibles/sources'
 import { addNftAssets, addNftTokens } from 'src/logic/collectibles/store/actions/addCollectibles'
 import { safeParamAddressFromStateSelector } from 'src/routes/safe/store/selectors'
 
+let isFetchingData = false
```

Comment:
Why did you decide to go with this approach instead of refactoring the actual users of this action which call it inside an interval?

This one, for example:
```
/src/routes/safe/container/hooks/useCheckForUpdates.tsx
```

Reply:
You can just return a promise from the action and wait until it is complete

Reply:
https://stackoverflow.com/questions/54054887/redux-how-to-call-an-action-and-wait-until-it-is-resolved

Reply:
From my point of view, the action is perfectly fine, the problem is how we use it

Reply:
Because it's simpler, returning multiple then's it's always more difficult to follow and could lead into callback hell's I think the current approach is easy to understand

Reply:
Ok, what happens if you try to call the action for two different safes? How is this going to work with global `isFetching` variable?

Reply:
Sorry this is not how actions are meant to work

Reply:
Shouldn't be a big problem, will just wait until the last fetch to finish and then once the second interval tick it will call again the function

Reply:
No, this a big problem. Abstract from the current implementation (inside an interval) and imagine that this is not inside an interval

Reply:
What does one do if he wants to fetch an ether balance for a different safe, maybe inside some modal? How does one do that keeping in mind that this action is being run in an interval?

Reply:
This is not the solution, this is just dealing with the consequences. The problem is how we use actions, not the action itself. Please fix the actual problem

Reply:
When implementing a solution, it may be helpful to also think what problems the solution can create and how long does it take for the problem to happen. With this solution, fetching data for two safes simultaneously is not possible, or fetching an info for one safe and then quickly dispatching an action for another. This is not some extraordinary use case, this can already be reproduced by user behaviour. So if we take this solution we will quickly have two problems: one, that the solution was meant to fix and one that it created

Reply:
What do you think now? it's basically the implementation that you suggested. I added promises for every action and now I wait until those promises finish before fetching again inside the interval